particles/Tutorial/Kotlin/README.md

```diff
@@ -0,0 +1,106 @@
+// path, We can assume the particle is implemented in kotlin.
+particle HelloWorldParticle in ''https://$arcs/bazel-bin/particles/Tutorial/Kotlin/hello_world.wasm''
+  // Notice the tab that starts this line. Whitespace matters
+  // in the Arcs manifest language, so this is very important.
+  // Don't worry about what this line does at the moment, we'll
+  // be getting to root and slots in more detail soon.
+  consume root
+
+// And now we are at the recipe definition!
+recipe HelloWorldRecipe
+  // Once again, notice the tab. This line tells us that the
+  // HelloWorldRecipe contains a HelloWorldParticle.
+  HelloWorldParticle
+  // Finally, we provide a human readable description for the
+  // recipe. This is optional, but will make it much easier and
+  // nicer to debug. Once again, notice the tab.
+  description `Tutorial 1: Hello World`
+```
+
+Alright, we’ve got our Arcs manifest file set. Now onto the Kotlin. Just like the .arcs file, this should be in your HelloWorld folder. We set the file name in the .arcs file above to be `HelloWorld.kt`.
+
+```kotlin
+// We start with the required imports and package info
+package arcs
+import kotlin.native.internal.ExportForCppRuntime
+
+// We define the HelloWorldParticle class to be a child
+// of the Particle class.
+class HelloWorldParticle : Particle() {
+
+  // For the moment, don't worry about this function. We will be explaining
+  // handles, slots, and root in plenty of details in upcoming tutorials.
+  override fun onHandleUpdate(handle: Handle) {
+    renderSlot("root")
+  }
+
+  // Again, don't worry about this function. We promise you'll understand
+  // what it does soon enough.
+  override fun onHandleSync(handle: Handle, allSynced: Boolean) {
+    if(allSynced) {
+      log("All handles synched\n")
+      renderSlot("root")
+    }
+  }
+
+  // Setup the console to log.
+    private fun console(s: String) {
+      log(s)
+    }
+
+  // Set the template to be "Hello, world!" as this is what we want
+  // to display.
+  override fun getTemplate(slotName: String): String {
+      return """<b>Hello, world!</b>"""
+  }
+}
+
+// To run Kotlin particles on the web, we have to convert them to
+// Web Assembly (WASM). Here is the boilerplate to ensure everything
+// gets linked properly.
+@Retain
+@ExportForCppRuntime("_newHelloWorldParticle")
+fun constructHelloWorldParticle(): WasmAddress {
+    log("_newHelloWorldParticle called")
+    return HelloWorldParticle().toWasmAddress()
+}
+```
+
+Finally, we need a 'BUILD' file. Arcs particles can be built using Bazel rules. Here's an example Bazel BUILD file for HelloWorld:
+
+```BUILD
+// Required imports
+load("//build_defs/kotlin_native:build_defs.bzl", "kt_wasm_binary", "kt_wasm_library")
+load("//build_defs:build_defs.bzl", "arcs_manifest", "arcs_kt_particle", "arcs_kt_wasm_binary")
+
+// Arcs Kotlin particle build rule
+arcs_kt_particle(
+    name = "hello_world_lib",
+    srcs = ["1_HelloWorld/HelloWorld.kt"],
```

Comment:
Is there a reason why we would recommend putting a BUILD file in a parent directory to the code?